File: StackQueue/주식가격.java
import java.util.Stack;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> stk = new Stack<>();
        stk.push(0);
        for(int i=1; i<prices.length; i++){
            while(!stk.isEmpty() && prices[stk.peek()] > prices[i]){
                answer[stk.peek()] = i - stk.peek();
                stk.pop();
            }
            stk.push(i);
        }
        while(!stk.isEmpty()){
            answer[stk.peek()] = prices.length - stk.peek() - 1;
            stk.pop();
        }
        return answer;
    }
}

File: stackqueue/주식가격.cpp
#include <string>
#include <vector>
#include <stack>

using namespace std;

vector<int> solution(vector<int> prices) {
    vector<int> answer(prices.size(), 0);
    stack<int> s;
    s.push(0);
    for(int i = 1; i < prices.size(); i++) {
        while(!s.empty() && prices[s.top()] > prices[i]) {
            answer[s.top()] = i - s.top();
            s.pop();
        }
        
        s.push(i);
    }
    
    while(!s.empty()){
        answer[s.top()] = prices.size() - 1 - s.top();
        s.pop();
    }
    
    return answer;
}